### ISSUE TITLE
Renderable locks fail with 2.10

### ISSUE THREAD

**Issue by franz** [buildbot/buildbot#5757]
Hello,

I have a buildbot setup with a bunch of DockerLatentWorkers and I use a renderable lock to avoid launching too much Docker builds on a physical machine. This worked fine with 2.8.3 but stopped working when i updated buildbot to 2.10. The error reported is:

```
2021-01-18 01:56:45+0000 [-] added buildset 182 to database
2021-01-18 01:56:45+0000 [-] from maybeStartBuild for builder 'llvm_100_pthread'
	Traceback (most recent call last):
	  File "/home/alarm/VENV/lib/python3.9/site-packages/buildbot/process/buildrequestdistributor.py", line 194, in popNextBuild
	    canStart = yield self.canStartBuild(worker, breq)
	  File "/home/alarm/VENV/lib/python3.9/site-packages/buildbot/process/buildrequestdistributor.py", line 267, in canStartBuild
	    return self.bldr.canStartBuild(worker, breq)
	  File "/usr/lib/python3.9/site-packages/twisted/internet/defer.py", line 1613, in unwindGenerator
	    return _cancellableInlineCallbacks(gen)
	  File "/usr/lib/python3.9/site-packages/twisted/internet/defer.py", line 1529, in _cancellableInlineCallbacks
	    _inlineCallbacks(None, g, status)
	--- <exception caught here> ---
	  File "/home/alarm/VENV/lib/python3.9/site-packages/buildbot/process/buildrequestdistributor.py", line 478, in _activityLoop
	    yield self._maybeStartBuildsOnBuilder(bldr)
	  File "/home/alarm/VENV/lib/python3.9/site-packages/buildbot/process/buildrequestdistributor.py", line 495, in _maybeStartBuildsOnBuilder
	    worker, breqs = yield bc.chooseNextBuild()
	  File "/home/alarm/VENV/lib/python3.9/site-packages/buildbot/process/buildrequestdistributor.py", line 62, in chooseNextBuild
	    worker, breq = yield self.popNextBuild()
	  File "/home/alarm/VENV/lib/python3.9/site-packages/buildbot/process/buildrequestdistributor.py", line 194, in popNextBuild
	    canStart = yield self.canStartBuild(worker, breq)
	  File "/usr/lib/python3.9/site-packages/twisted/internet/defer.py", line 1418, in _inlineCallbacks
	    result = g.send(result)
	  File "/home/alarm/VENV/lib/python3.9/site-packages/buildbot/process/builder.py", line 300, in canStartBuild
	    locks = yield props.render(locks)
	builtins.AttributeError: 'NoneType' object has no attribute 'render'
```

I'll create a PR with a fix.

**Comment by franz** [comment]
Ok, so this simple patch on `buildbot/process/builder.py`

```
         def setupPropsIfNeeded(props):
             if props is not None:
-                return None
+                return props
             props = Properties()
```

... fixes the issue on my buildbot instance, but running `trial` shows failures. I don't have the time to debug this further ATM, so if someone else can confirm this is the correct fix, that would be great.

**Comment by p12tic** [comment]
Thanks a lot for the bug report. This will be fixed in #5782 and backported to 2.10.1.
